core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -228,6 +260,40 @@ private[spark] class EventLoggingListener(
     fileSystem.rename(new Path(logPath + IN_PROGRESS), target)
   }
 
+  /**
+   * According to the updated event to modify the maintained event's metrics
+   * @param executorId  the executor whose metrics will be modified
+   */
+  private def updateModifiedMetrics(executorId: String): Unit = {
+    val toBeModifiedEvent = executorIdToModifiedMaxMetrics.get(executorId)
+    val latestEvent = executorIdToLatestMetrics.get(executorId)
+    if (toBeModifiedEvent.isEmpty) {
+      if (latestEvent.isDefined) executorIdToModifiedMaxMetrics.update(executorId, latestEvent.get)
+    } else {
+      val toBeModifiedMetrics = toBeModifiedEvent.get.executorMetrics.transportMetrics
+      if (toBeModifiedMetrics.isDefined) {
```

Comment:
won't `latestEvent` always be defined?  In fact, at the one call site, you could even just pass in `latestEvent` directly so you avoid another lookup.  I also think this becomes slightly cleaner w/ pattern matching:

```scala
private def updateModifiedMetrics(executorId: String, latestEvent: SparkListenerExecutorMetricsUpdate): Unit = {
  executorIdToModifiedMaxMetrics.get(executorId) match {
    case None => executorIdToModifiedMaxMetrics.update(executorId, latestEvent)
    case Some(toBeModEvent) =>
      val toBeModMetrics = toBeModEvent.executorMetrics.transportMetrics
      ...
  }
}
```

and depending on whether or not we need to keep `tranportMetrics` as an option, we may need to handle the else case here, right?